examples/chat/README.md

```diff
@@ -0,0 +1,43 @@
+# p2p chat app with libp2p
+
+This program demonstrates a simple p2p chat application. It can work between two peers if
+1. Both have private IP address (same network).
+2. At least one of them has a public IP address.
+
+Assume if 'A' and 'B' are on different networks host 'A' may or may not have a public IP address but host 'B' has one.
+
+Usage: Run `./chat -sp <SOURCE_PORT>` on host 'B' where <SOURCE_PORT> can be any port number. Now run `./chat -d <MULTIADDR_B>` on node 'A' [<MULTIADDR_B> is multiaddress of host 'B' which can be obtained from host 'B' console].
+
+## Build
+
+First install libp2p using `go get -d github.com/libp2p/go-libp2p/...`
```

Comment:
We generally manage our dependencies using a tool called `gx` (to avoid random breakage when our dependencies change). This should probably say:

> To build the example, first run `make deps` in the root directory.